### DISCUSSION TITLE
Search for POI not ordered by distance

### DISCUSSION THREAD

**Discussion by Clipper11** [osmandapp/OsmAnd#16344] [General]
Hi,

If have a question to the search order lineup in osmand version 4.3.8
I am in Germany, 73116 Wäschenbeuren and I am searching for 'Wertstoffhof' (recycling station)
If I am doing that, the nearest poi will not shown as first. This is confusing. Can I influence that in the settings, or is this an error?
The nearest is 290 meter, but listed not as first. Screenshot attached

Thanks for your support.
best regards
Marius

![Screenshot_20230129_121633_OsmAnd](https://user-images.githubusercontent.com/123870871/215322749-fcc0a116-bf36-45f9-8abc-8eb70ff906a1.jpg)

**Comment by vshcherb** [comment]
That's done to prioritize precise name matches which in some situations are more important - https://github.com/osmandapp/OsmAnd/issues/13797

**Reply by mariush444** [reply]
but it is still unclear what priority is. When someone just look at this it seems to be chaos.

just ideas:

1.  just put the result in simple table with column name and distance then when someone tap on the name then collating is by name (more important prioritize precise name matches) when user tap on the distance then collating is by distance.

Alternative
2 switch  (name<->distance) before/beside search button/icon.

PS
I think that fighting with some exclusions, unique cases we loose simplicity.
definitely it was happen with navigation. Calculating of the 300 km route takes more than 20 minutes. Other apps makes that in 3-4 seconds. Probably osmand try to optimize the route to be shorter for 1 kilometer or faster for 5 minutes but it makes osmand useless. 
but please don't get me wrong I don't want to criticize osmand team. I just think that sometimes it is better to don't to much optimization when we lost simplicity, time or usefulness. **just wondering**

**Reply by vshcherb** [reply]
300 km more than 20 minutes sounds a bit dramatic, on our test devices car navigation calculates 1-2 min - 1000 km, if it takes longer it should just crash and should be checked.
- https://github.com/osmandapp/OsmAnd/issues/16397 - there is ongoing discussion here as well for reference.

**Reply by mariush444** [reply]
look at your comments please 
![image](https://user-images.githubusercontent.com/66887280/217320980-9095a74c-3d12-451c-9b97-e05c20c96c05.png)
![image](https://user-images.githubusercontent.com/66887280/217322758-2b1dcc66-19b6-4fbf-9dea-3ff717c09820.png)


can you show how your navi calculates this case in 2 minutes, please?

start point 49.00537 8.40449
end point 52.10229 7.40721
Graphhopper does it in less than 5 seconds.

**Reply by vshcherb** [reply]
I don't think we should continue routing discussion here it only makes it's offtopic - **just for the record** this route takes 427 km - 1 300 000 (> 1 000 000 that considered reasonable) - ~3 min calculation S21,   and it's not related that OSRM algorithms could do in 1 seconds that A* algorithms are stuck)

**Reply by vshcherb** [reply]
Nothing wrong with the comments all 2 issues are related to this topic

**Reply by mariush444** [reply]
so maybe there is other problem.
My calculation take more than 30 minutes and don't finish with any result.
There is info about not loaded maps but there are loaded and "check the necessary maps" doesn't help.

What map do you use? I downloaded full and for second test I downloaded roads only. No positive result.

![routing](https://user-images.githubusercontent.com/66887280/218274397-632e1f27-0433-4df3-b623-148c2f1d09ba.jpg)
Maps are from februare 2023

**Reply by vshcherb** [reply]
Not enough memory you can put it higher in Development Settings but it might crash.

**Comment by mariush444** [comment]
How much memory does your S21 have?



-------- Wiadomość oryginalna --------
Od: ***@***.***
Wysłane: 14 lutego 2023 16:32:59 CET
Do: ***@***.***
DW: Mars Yush ***@***.***>, ***@***.***
Temat: Re: [osmandapp/OsmAnd] Search for POI not ordered by distance (Discussion #16344)

Not enough memory you can put it higher in Development Settings but it might crash.

-- 
Reply to this email directly or view it on GitHub:
https://github.com/osmandapp/OsmAnd/discussions/16344#discussioncomment-4972202
